distribution/tools/fips-demo-installer-cli/src/main/java/org/opensearch/tools/cli/fips/truststore/TrustStoreService.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tools.cli.fips.truststore;
+
+import java.nio.file.Path;
+import java.util.function.Function;
+
+import picocli.CommandLine.Model.CommandSpec;
+
+/**
+ * Service for managing trust store operations.
+ * Handles generation and configuration of FIPS-compliant trust stores.
+ */
+public class TrustStoreService {
+
+    private final UserInteractionService userInteraction;
+
+    public TrustStoreService(UserInteractionService userInteraction) {
+        this.userInteraction = userInteraction;
+    }
+
+    /**
+     * Generates a new BCFKS trust store from the JVM default trust store.
+     *
+     * @param spec the command specification for output
+     * @param options common command-line options
+     * @param confPath path to the OpenSearch configuration directory
+     * @return exit code (0 for success, 1 for failure)
+     */
+    public Integer generateTrustStore(CommandSpec spec, CommonOptions options, Path confPath) {
+        if (!userInteraction.confirmAction(spec, options, "Generate new BCFKS trust store from system defaults?")) {
+            spec.commandLine().getOut().println("Operation cancelled.");
+            return 0;
+        }
+
+        var password = userInteraction.promptForPasswordWithConfirmation(spec, options, "Enter trust store password");
+        spec.commandLine().getOut().println("Generating BCFKS trust store...");
+        var properties = Function.<Path>identity()
+            .andThen(path -> CreateFipsTrustStore.loadJvmDefaultTrustStore(spec, path))
+            .andThen(trustStore -> CreateFipsTrustStore.convertToBCFKS(spec, trustStore, options, password, confPath))
+            .andThen(bcfksPath -> CreateFipsTrustStore.configureBCFKSTrustStore(bcfksPath, password))
+            .apply(Path.of(System.getProperty("java.home")));
```

Comment:
`System.getProperty("java.home")` does not look right, OpenSearch could and often uses bundled JDK in the `jdk` folder, not the system installed one(s), `opensearch-env` [1] is a good place to look at how OpenSearch detects the JDK to run with.

[1] https://github.com/opensearch-project/OpenSearch/blob/main/distribution/src/bin/opensearch-env